ui/src/main/java/com/nextcloud/android/common/ui/theme/utils/MaterialViewThemeUtils.kt

```diff
@@ -65,39 +68,54 @@ class MaterialViewThemeUtils @Inject constructor(schemes: MaterialSchemes, priva
 
     fun themeFAB(fab: FloatingActionButton) {
         withScheme(fab) { scheme ->
-            fab.backgroundTintList = ColorStateList.valueOf(scheme.primaryContainer)
-            fab.imageTintList = ColorStateList.valueOf(scheme.onPrimaryContainer)
+            fab.backgroundTintList = ColorStateList(
+                arrayOf(
+                    intArrayOf(android.R.attr.state_enabled),
+                    intArrayOf(-android.R.attr.state_enabled)
+                ),
+                intArrayOf(
+                    scheme.primaryContainer,
+                    Color.GRAY
+                )
+            )
+
+            fab.imageTintList = ColorStateList(
+                arrayOf(
+                    intArrayOf(android.R.attr.state_enabled),
+                    intArrayOf(-android.R.attr.state_enabled)
+                ),
+                intArrayOf(
+                    scheme.onPrimaryContainer,
+                    Color.WHITE
+                )
+            )
```

Comment:
@AndyScherzinger you can use `buildColorStateList` to simplify this, for reference:

```suggestion
            fab.backgroundTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.primaryContainer,
                -android.R.attr.state_enabled to Color.GRAY
            )

            fab.imageTintList = buildColorStateList(
                android.R.attr.state_enabled to scheme.onPrimaryContainer,
                -android.R.attr.state_enabled to Color.WHITE
            )
```
@AndyScherzinger  you can use `buildColorStateList` to simplify